xdsl/dialects/llvm.py

```diff
@@ -564,6 +564,57 @@ def print(self, printer: Printer):
         printer.print(self.arg.type, " to ", self.res.type)
 
 
+class IntegerConversionOpNNeg(IRDLOperation, ABC):
+    arg = operand_def(IntegerType)
+    res = result_def(IntegerType)
+    traits = traits_def(NoMemoryEffect())
+    non_neg = opt_prop_def(UnitAttr, prop_name="nonNeg")
+
+    def __init__(
+        self,
+        arg: SSAValue,
+        res_type: Attribute,
+        attributes: dict[str, Attribute] = {},
+        non_neg: UnitAttr | None = None,
+    ):
+        super().__init__(
+            operands=[arg],
+            attributes=attributes,
+            result_types=[res_type],
+            properties={
```

Comment:
```suggestion
            operands=(arg,),
            attributes=attributes,
            result_types=(res_type,),
            properties={
```

Reply:
Using tuples is marginally faster but it adds up if done consistently thoughout the project

Reply:
Done, I'll also make a note to apply that to the rest of the file